src/main/java/org/betonquest/betonquest/quest/event/compass/CompassEvent.java

```diff
@@ -0,0 +1,110 @@
+package org.betonquest.betonquest.quest.event.compass;
+
+import org.betonquest.betonquest.BetonQuest;
+import org.betonquest.betonquest.api.QuestCompassTargetChangeEvent;
+import org.betonquest.betonquest.api.config.quest.QuestPackage;
+import org.betonquest.betonquest.api.logger.BetonQuestLogger;
+import org.betonquest.betonquest.api.profiles.Profile;
+import org.betonquest.betonquest.api.quest.event.Event;
+import org.betonquest.betonquest.exceptions.QuestRuntimeException;
+import org.betonquest.betonquest.quest.event.tag.AddTagChanger;
+import org.betonquest.betonquest.quest.event.tag.DeleteTagChanger;
+import org.betonquest.betonquest.quest.event.tag.TagEvent;
+import org.betonquest.betonquest.utils.Utils;
+import org.betonquest.betonquest.utils.location.CompoundLocation;
+import org.bukkit.Location;
+import org.bukkit.plugin.PluginManager;
+
+/**
+ * Event to set a compass target and manage compass points.
+ */
+public class CompassEvent implements Event {
+
+    /**
+     * Custom {@link BetonQuestLogger} instance for this class.
+     */
+    private final BetonQuestLogger log;
+
+    /**
+     * BetonQuest instance to use to get the event factory.
+     */
+    private final BetonQuest betonQuest;
+
+    /**
+     * Plugin manager to use to call the event.
+     */
+    private final PluginManager pluginManager;
+
+    /**
+     * The action to perform on the compass.
+     */
+    private final CompassTargetAction action;
+
+    /**
+     * The compass point to set.
+     */
+    private final String compass;
+
+    /**
+     * The location to set the compass to.
+     */
+    private final CompoundLocation compassLocation;
+
+    /**
+     * The quest package to use for logging.
+     */
+    private final QuestPackage questPackage;
+
+    /**
+     * Create the compass event.
+     *
+     * @param log             the logger
+     * @param betonQuest      the BetonQuest instance
+     * @param pluginManager   the plugin manager
+     * @param action          the action to perform
+     * @param compass         the compass point
+     * @param compassLocation the location to set the compass to
+     * @param questPackage    the quest package
+     */
+    public CompassEvent(final BetonQuestLogger log, final BetonQuest betonQuest, final PluginManager pluginManager,
+                        final CompassTargetAction action, final String compass, final CompoundLocation compassLocation,
+                        final QuestPackage questPackage) {
+        this.log = log;
+        this.betonQuest = betonQuest;
+        this.pluginManager = pluginManager;
+        this.action = action;
+        this.compass = compass;
+        this.compassLocation = compassLocation;
+        this.questPackage = questPackage;
+    }
+
+    @Override
+    public void execute(final Profile profile) throws QuestRuntimeException {
```

Comment:
When using this approach I suggest de7121622a1bf77bc37b4df9130f6dc5123b6f84 as code change.